test/test_python_dispatch.py

```diff
@@ -79,6 +80,117 @@ def my_sum(*args, **kwargs):
         # Validate that the old behavior is restored for sum
         self.assertEqual(torch.sum(x), torch.tensor(3))
 
+    def test_override_cuda_with_jiterator(self) -> None:
+        def override_where_cuda() -> None:
+            # Example 1: Invert the behavior of where's condition input
+            not_where_code_string = '''
+            template <typename T> T inverted_where(bool cond, T a, T b){
+                return !cond ? a : b;
+            }
+            '''
+            jitted_where = _create_jit_fn(not_where_code_string)
+
+            CALLED = [False]
+
+            def inverted_where(*args, **kwargs):
+                CALLED[0] = True
+                return jitted_where(*args, **kwargs)
+
+            # overriding where's cuda kernel with Jiterator generated kernel
+            my_lib = Library("aten", "IMPL")
+            my_lib.impl('aten::where.self', inverted_where, "CUDA")
+
+            device = 'cuda'
+            cond = torch.tensor([True, True, False], device=device, dtype=torch.bool)
+            x = torch.tensor([1, 2, 3], device=device)
+            y = torch.tensor([-1, -2, -3], device=device)
+
+            self.assertEqual(torch.where(cond, x, y), torch.tensor([-1, -2, 3]))
+            self.assertTrue(CALLED[0])
+            del my_lib
```

Comment:
I guess it's nice to have explicit deletes but they are not necessary

Reply:
yep.

Reply:
add a check that the old behavior is retained after your library is deleted